draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1504 @@
+
+
+def jy_per_beam_to_kelvin(freq, bl_length):
+    """Conversion factor from Jy/beam to kelvin unit.
+
+    The conversion factor is C = (10**-26 * lambda**2)/(2 * K_boltzmann * omega_PSF),
+    where omega_PSF is the beam area in sr unit.
+
+    Parameters
+    ----------
+    freq : np.ndarray[freq]
+        frequency in MHz unit
+    bl_length : float
+        baseline length in meter
+
+    Returns
+    -------
+    C : np.ndarray[freq]
+     The conversion factor from Jy/beam to Kelvin
+    """
+    Jy = 1.0e-26  # W m^-2 Hz^-1
+    wl = units.c / (freq * 1e6)  # freq of the map in MHz
+
+    # Estimate the PSF area, assuming a Gaussian PSF, based on max baseline
+    # Bmaj=Bmin= PSF_arcsec; beam_area = (pi * Bmaj * Bmin)/(4 * log(2))
+    PSF = 1.22 * wl / bl_length
+    PSF = np.degrees(PSF)  # in deg
+    omega_psf = (np.pi * PSF**2) / (4 * np.log(2))
+    omega_psf_sr = omega_psf * (np.pi / 180.0) ** 2  # convert the PSF area to sr
+
+    kB = units.k_B  # Boltzmann Const in J/k (1.38 * 10-23)
+    return wl**2 * Jy / (2 * kB * omega_psf_sr)
+
+
+def noise_equivalent_bandwidth(N, window):
+    """Calculate the relative equivalent noise bandwidth of window function.
+
+    Following this:
+    https://dsp.stackexchange.com/questions/70273/find-the-equivalent-noise-bandwidth
+
+    Parameters
+    ----------
+    N: int
+     Size of the window
+    window : array_like
+        A 1-Dimenaional array like.
+
+    Returns
+    -------
+    float
+    """
+    fsel = np.arange(N)
+    x = fsel / N
+    w = tools.window_generalised(x, window=window)
+
+    return np.sum(w) ** 2 / (np.sum(w**2) * len(w))
+
+
+def get_fourier_modes(ra, dec, delays, redshift, cosmo=None):
+    """Compute the spatial and line-of-sight Fourier modes.
+
+    Parameters
+    ----------
+    ra : np.array[nra]
+      The RA axis in deg unit.
+    dec : np.narray[ndec]
+      The DEC axis in deg unit.
+    delays : np.array[ndelay]
+      The delay axis in second unit.
+    redshift : float
+      redshift at the centre of the band.
+    cosmo: Cosmology object
+        Default is cora.util.Cosmology() default.
+
+    Returns
+    -------
+    kx : np.array[nra]
+      The Fourier modes conjugate to RA axis, unit [h/Mpc].
+    ky : np.array[ndec]
+      The Fourier modes conjugate to DEC axis, unit [h/Mpc].
+    u : np.array[nra]
+      gridded u-coordinates
+    v : np.array[nel]
+      gridded v-coordinates
+    k_parallel : np.array[ndelay]
+      The Fourier modes conjugate to frequency axis, unit [h/Mpc].
+    """
+    if cosmo is None:
+        cosmo = get_cosmo()
+
+    nra = ra.size
+    ndec = dec.size
+
+    # Mean resolution in RA and DEC and convert that to radian
+    res_ra_radian = np.deg2rad(np.mean(np.diff(ra)))
+    res_dec_radian = np.deg2rad(np.mean(np.diff(dec)))
+
+    DMz = dRperp_dtheta(redshift, cosmo=cosmo)  # in [h^-1 Mpc]
+
+    # Convert the RA and DEC resolution to Mpc unit
+    d_RA_Mpc = DMz * res_ra_radian
```

Comment:
```suggestion
    d_RA_Mpc = DMz * res_ra_radian * np.mean(np.cos(np.deg2rad(dec)))
```
I think you need a factor of cos(dec) in the conversion here.

Reply:
I added the cos(dec) term